src/rgw/rgw_auth_s3.cc

```diff
@@ -446,7 +452,11 @@ static inline int parse_v4_auth_header(const req_info& info,               /* in
   }
 
   if (info.env->exists("HTTP_X_AMZ_SECURITY_TOKEN")) {
-    sessiontoken = info.env->get("HTTP_X_AMZ_SECURITY_TOKEN");
+    boost::optional<const std::string&> token;
+    token = info.env->get_optional("HTTP_X_AMZ_SECURITY_TOKEN");
```

Comment:
```suggestion
    auto token = info.env->get_optional("HTTP_X_AMZ_SECURITY_TOKEN");
```

Reply:
should we replace the data type `boost::optional<const std::string&>` with `auto` across all the changes to be consistent?

Reply:
@yuvalif Why use auto instead?

Reply:
it is walwasy better to define and set a variable at the same time, using the "auto" keyword is just a side benefit from that (shorter code), but if you prefer to be explicitm, you can do:
```cpp
 boost::optional<const std::string&> token = info.env->get_optional("HTTP_X_AMZ_SECURITY_TOKEN");
```

Reply:
Thanks for sharing! Let's go with `auto`